### DISCUSSION TITLE
Reason for longer time to import on the first execution

### DISCUSSION THREAD

**Discussion by thomwolf** [pyvista/pyvista#2573] [Q&A] [answered]
Hi,
I was wondering what was the underlying reason that pyvista takes a longer time to import on its first execution.
What is happening under the hood that take some time?

**Comment by akaszynski** [comment, marked as answer]
PyVista, has to load several vtk shared libraries on initialization. While we've reduced or lazily imported many of those, it still can take some time (especially on windows). You can read more about in https://github.com/pyvista/pyvista/pull/1156.

If you want to compare the import time with just vanilla vtk, try evaluating the initial import of `vtk` vs just `pyvista`. You should see some improvement when importing just `pyvista` because we're selectively importing a subset of `vtkmodules`.

**Comment by thomwolf** [comment]
Ok, thanks a lot for work on #1156 (and #1163 and #1029). It's great to see you take into account this aspect of the user experience.

I still notice that the first time I import `pyvista` in a blank new env where pyvista is newly installed, the first import take quite longer that subsequent import time (in subsequent python sessions) but from walking in it step by step it seems to be something happening inside vtk, the first time it is imported after installation.

I'll post here if I ever get down to the root of it.
